apps/web/src/app/[orgShortcode]/settings/user/security/page.tsx

```diff
@@ -581,3 +620,180 @@ export default function Page() {
     </div>
   );
 }
+
+function DeleteAccount() {
+  const router = useRouter();
+  const { data: accountData, isLoading: accountDataLoading } =
+    platform.account.security.deleteAccountPre.useQuery(void 0, {});
+
+  const confirmationFormSchema = z.object({
+    username: z.literal(accountData?.username, {
+      errorMap: () => ({ message: 'Must match your username exactly' })
+    })
+  });
+
+  const form = useForm<z.infer<typeof confirmationFormSchema>>({
+    resolver: zodResolver(confirmationFormSchema),
+    defaultValues: {
+      username: ''
+    }
+  });
+
+  const validated = form.formState.isValid;
+
+  const { mutate: deleteAccountConfirm, isPending: deletingAccount } =
+    platform.account.security.deleteAccountConfirm.useMutation({
+      onSuccess: () => {
+        router.push(`/`);
```

Comment:
use `window.location.replace('/')`, it reloads the page, so we would not need to care about query cache and stuff and also calls middleware which removes the expired cookie